src/macros.h

```diff
@@ -56,20 +56,20 @@
     Nan::ThrowError(#var " must be a number"); \
     return; \
   } \
-  unsigned int var = name->Uint32Value();
+  int64_t var = name->IntegerValue();
 
 #define ASSERT_UINT_BOUNDS(name, var, min, max) \
   if (!name->IsNumber()) { \
     Nan::ThrowError(#var " must be a number"); \
     return; \
   } \
-  unsigned int var = name->Uint32Value(); \
+  int64_t var = name->IntegerValue(); \
   \
-  if (min > 0 && var < min) { \
+  if (var < min) { \
     Nan::ThrowError(#var " must be at least " #min); \
     return; \
   } \
-  if (max <= sizeof(unsigned int) && var > max) { \
+  if ((uint64_t) var > max) { \
```

Comment:
This will work, but is rather peculiar. Negative value of `var` will roll up to very big `uint64_t` value (0x8000000000000000), so the error will happen. Should we check if `var` is positive before this?